### ISSUE TITLE
TODO: create publish/release workflow for JSR and NPM

### ISSUE THREAD

**Issue by nberlette** [nberlette/typedefs#2]
The new monorepo will need a publishing workflow for both NPM and JSR.

- [ ] Run CI/CD workflow to ensure code is formatted / linted / tested well

I'm creating a custom fork of the `@deno/bump-workspaces` tool, which will do the following:

- [ ] Assess all new commits to determine each package's release type from the commit messages.
- [ ] Increment the **root** `deno.json` file version with the **highest** release type found across all of the workspace packages.
- [ ] Increment all workspace versions to match the root `deno.json`
- [ ] Commit the changes to a release branch
- [ ] Draft a new pull request for the release with auto-generated details

At this point, the pull request must be manually reviewed and merged.

Once it is merged, another workflow (or a another job in the same workflow) will...

- [ ] Tag/commit/push to main. (this publishes to `deno.land/x/typedefs`)
- [ ] Run `deno publish` or `npx jsr publish` (for each package)
- [ ] Build each package with `dnt` for NPM
- [ ] Upload an artifact of the results from the build
- [ ] Authenticate and publish to NPM (under the @type2 org)
- [ ] Authenticate and publish to GPR (GitHub Package Registry)
